src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/User/Views.ts

```diff
@@ -352,12 +358,36 @@ export var metaDirective = (adhConfig : AdhConfig.IService, adhResourceArea : Ad
     };
 };
 
+export class BadgeAssignment {
+    constructor(
+        private title : string,
+        private description : string,
+        private name : string
+    ) {}
+}
+
+var getBadges = (
+    adhHttp : AdhHttp.Service<any>,
+    $q : angular.IQService
+) => (user : RIUser) : angular.IPromise<BadgeAssignment[]> => {
+    return $q.all(_.map(user.data[SIBadgeable.nick].assignments, (assignmentPath : string) => {
+        return adhHttp.get(assignmentPath).then((assignment : RIBadgeAssignment) => {
+            var description = assignment.data[SIDescription.nick].description;
+            return adhHttp.get(assignment.data[SIBadgeAssignment.nick].badge).then((badge) => {
+                var title = badge.data[SITitle.nick].title;
+                var name = badge.data[SIName.nick].name;
+                return new BadgeAssignment(title, description, name);
+            });
+        });
+    }));
+};
```

Comment:
This code is duplicated in `MeinBerlin/Kiezkassen/Proposal/Proposal.ts` and in `MercatorProposal/MercatorProposal.ts`. Please move it to a single place.